WordPress/JetpackStatsWidgets/Remote service/StatsWidgetsService.swift

```diff
@@ -176,3 +174,16 @@ private extension Date {
         return addingTimeInterval(delta)
     }
 }
+
+private extension StatsWidgetsService {
+    private func createStatsService(for widgetData: HomeWidgetData) throws -> StatsServiceRemoteV2 {
+        let token = try SFHFKeychainUtils.getPasswordForUsername(AppConfiguration.Widget.Stats.keychainTokenKey,
+                                                                 andServiceName: AppConfiguration.Widget.Stats.keychainServiceName,
+                                                                 accessGroup: WPAppKeychainAccessGroup)
+        let wpApi = WordPressComRestApi(oAuthToken: token)
+        let service = StatsServiceRemoteV2(wordPressComRestApi: wpApi, siteID: widgetData.siteID, siteTimezone: widgetData.timeZone)
```

Comment:
If it retains all these values for the lifetime of the widgets, can it become a problem?

P.S. It's a significant design flaw that you have to _initialize_ a class with an auth token 😒

Reply:
That's a fair question. This service continues to live in the memory and is only replaced when another refresh begins. 🤔

The quick solution I could think of is this https://github.com/wordpress-mobile/WordPress-iOS/pull/23088/commits/19d635dd8166cc01e2e11877459c519cfb358892, to make sure that StatsService is created, retained, and then released during each service call. Although the produced solution is pretty messy.

Reply:
Yeah, we should really revisit the `WordPressComRestApi`. I raised it before – the token should _not_ be in the initializer and should be assigned on each call.

Retaining for the lifetime of the call sounds OK, but, again – I think it's another flaw in `WordPressComRestApi` design. I'd expect it to retain itself until it completes the calls.